File: src/main/java/org/is/bandmanager/service/CoordinatesService.java
package org.is.bandmanager.service;

import jakarta.validation.Valid;
import org.is.bandmanager.dto.CoordinatesDto;
import org.is.bandmanager.dto.request.CoordinatesRequest;
import org.is.bandmanager.model.Coordinates;

import java.util.List;


public interface CoordinatesService {

    CoordinatesDto create(@Valid CoordinatesRequest request);

    List<CoordinatesDto> getAll();

    CoordinatesDto get(Long id);

    Coordinates getEntity(Long id);

    CoordinatesDto update(Long id, @Valid CoordinatesRequest request);

    CoordinatesDto delete(Long id);

}

File: src/app/services/coordinates.service.ts
import {Injectable} from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {Observable} from 'rxjs';
import {Coordinates} from '../models/coordinates.model';
import {CoordinatesRequest} from '../models/requests/coordinates-request.model';

@Injectable({
  providedIn: 'root'
})
export class CoordinatesService {
  private baseUrl = '/coordinates';

  constructor(private http: HttpClient) {
  }

  getAll(): Observable<Coordinates[]> {
    return this.http.get<Coordinates[]>(this.baseUrl);
  }

  get(id: number): Observable<Coordinates> {
    return this.http.get<Coordinates>(`${this.baseUrl}/${id}`);
  }

  create(request: CoordinatesRequest): Observable<Coordinates> {
    return this.http.post<Coordinates>(this.baseUrl, request);
  }

  update(id: number, request: CoordinatesRequest): Observable<Coordinates> {
    return this.http.put<Coordinates>(`${this.baseUrl}/${id}`, request);
  }

  delete(id: number): Observable<Coordinates> {
    return this.http.delete<Coordinates>(`${this.baseUrl}/${id}`);
  }

}
